[inductor] add support for TRITON_INTERPRET

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #140838
* __->__ #140841

Was debugging the issue lower in the stack and found this to be helpful / quick enough to add.

Fix for https://github.com/pytorch/pytorch/issues/123956

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

eellison committed:
[WIP] add support for TRITON_INTERPRET

[ghstack-poisoned]

eellison committed:
Update on "[WIP] add support for TRITON_INTERPRET"


Was debugging the issue lower in the stack and found this to be helpful / quick enough to add.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

exclamaforte commented on torch/_inductor/async_compile.py:
```diff
@@ -212,6 +212,11 @@ def triton(self, kernel_name: str, source_code: str, device_str: str = "cuda"):
         _compile_start()
         _set_triton_ptxas_path()
 
+        if os.environ.get("TRITON_INTERPRET", "0") == "1":
```
Might be better to be in a config so people can see it as an option.

eellison replied:
You need it to be set as an envvar otherwise triton wont return the interpreted function. So setting the config value wouldnt be enough wed need to thread it to triton as well.. going to leave like this for now

eellison committed:
Update on "[inductor] add support for TRITON_INTERPRET"


Was debugging the issue lower in the stack and found this to be helpful / quick enough to add.

Fix for https://github.com/pytorch/pytorch/issues/123956

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

eellison commented:
@pytorchbot merge

eellison committed:
Update on "[inductor] add support for TRITON_INTERPRET"


Was debugging the issue lower in the stack and found this to be helpful / quick enough to add.

Fix for https://github.com/pytorch/pytorch/issues/123956

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

eellison commented:
@pytorchbot merge

eellison committed:
Update on "[inductor] add support for TRITON_INTERPRET"


Was debugging the issue lower in the stack and found this to be helpful / quick enough to add.

Fix for https://github.com/pytorch/pytorch/issues/123956

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]
